bcda/models/fhir/alr/observation.go

```diff
@@ -0,0 +1,156 @@
+			exclusionFields[k] = v
+		} else if riskFlagsPattern.MatchString(k) {
+			riskFlagFields[k] = v
+		} else if riskScoresPattern.MatchString(k) {
+			riskScoreFields[k] = v
+		}
+	}
+
+	return []*fhirmodels.Observation{
+		assignment(mbi, assignmentFields),
+		enrollment(mbi, enrollmentFields),
+		exclusion(mbi, exclusionFields),
+		riskFlags(mbi, riskFlagFields),
+		riskScores(mbi, riskScoreFields),
+	}
+}
+
+func assignment(mbi string, keyValue map[string]string) *fhirmodels.Observation {
+	observation := &fhirmodels.Observation{}
+	observation.Code = codeableConcept("Assignment", "Assignment flags, step, newly assigned, etc.")
+	observation.Subject = &fhirdatatypes.Reference{Identifier: mbiIdentifier(mbi)}
+	observation.Component = observationComponents(assignmentPattern, "assignment", keyValue)
+	return observation
+}
+
+func enrollment(mbi string, keyValue map[string]string) *fhirmodels.Observation {
+	observation := &fhirmodels.Observation{}
+	observation.Code = codeableConcept("Enrollment", "Monthly enrollment flags")
+	observation.Subject = &fhirdatatypes.Reference{Identifier: mbiIdentifier(mbi)}
+	observation.Component = observationComponents(enrollmentPattern, "enrollment", keyValue)
+	return observation
+}
+
+func exclusion(mbi string, keyValue map[string]string) *fhirmodels.Observation {
+	observation := &fhirmodels.Observation{}
+	observation.Code = codeableConcept("Exclusion", "Exclusion reasons")
+	observation.Subject = &fhirdatatypes.Reference{Identifier: mbiIdentifier(mbi)}
+	observation.Component = observationComponents(exclusionPattern, "exclusion", keyValue)
+	return observation
+}
+
+func riskFlags(mbi string, keyValue map[string]string) *fhirmodels.Observation {
+	observation := &fhirmodels.Observation{}
+	observation.Code = codeableConcept("Risk Flags", "HCC risk flags")
+	observation.Subject = &fhirdatatypes.Reference{Identifier: mbiIdentifier(mbi)}
+	observation.Component = riskFlagObservationComponents(keyValue)
+	return observation
+}
+
+func riskScores(mbi string, keyValue map[string]string) *fhirmodels.Observation {
+	observation := &fhirmodels.Observation{}
+	observation.Code = codeableConcept("Risk Scores", "HCC and other risk scores")
+	observation.Subject = &fhirdatatypes.Reference{Identifier: mbiIdentifier(mbi)}
+	observation.Component = observationComponents(exclusionPattern, "risk_scores", keyValue)
+	return observation
+}
+
+func codeableConcept(code, display string) *fhirdatatypes.CodeableConcept {
+	return &fhirdatatypes.CodeableConcept{
+		Coding: []*fhirdatatypes.Coding{
+			{System: fhirURI("http://cms.gov/CodeSystem/alr"),
+				Code: &fhirdatatypes.Code{Value: "ALR"},
+			},
+			{System: fhirURI("http://cms.gov/CodeSystem/alr"),
+				Code:    &fhirdatatypes.Code{Value: code},
+				Display: fhirString(display),
+			},
+		},
+	}
+}
+
+func observationComponents(pattern *regexp.Regexp, system string, keyValue map[string]string) []*fhirmodels.Observation_Component {
```

Comment:
Is pattern going to be used later on?

Reply:
I'm thinking this gives us the flexibility to encode N number of observation resources. What're your thoughts on it?